core/templates/dev/head/pages/library/Library.js

```diff
@@ -80,6 +80,36 @@ oppia.controller('Library', [
       $http.get('/libraryindexhandler').success(function(data) {
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
+        $scope.userIsLoggedIn = GLOBALS.userIsLoggedIn;
+        var activityDetails = {};
+        if ($scope.userIsLoggedIn) {
+          $scope.activitiesOwned = {explorations: {}, collections: {}};
+          $http.get('/creatordashboardhandler/data')
+            .then(function(response) {
+              $scope.libraryGroups.forEach(function(activityGroups) {
+                var activity = activityGroups.activity_summary_dicts;
+
+                activity.forEach(function(activity) {
+                  $scope.activitiesOwned.explorations[activity.id] = false;
+                  $scope.activitiesOwned.collections[activity.id] = false;
```

Comment:
Given a particular activity, why are we treating it as both an exploration and a collection?

Reply:
I took care of it. Thanks!